syntax = "proto3";

message Vector3 {
  float x = 1;
  float y = 2;
  float z = 3;
}

message Quaternion {
  float w = 1;
  float x = 2;
  float y = 3;
  float z = 4;
}

message FusedSensorData {
  uint64 timestamp_us = 1; // Timestamp in microseconds
  Quaternion attitude_q = 2; // Attitude as a quaternion
  Vector3 position_m = 3;   // Position in meters
  Vector3 velocity_mps = 4; // Velocity in meters per second
  Vector3 acceleration_mps2 = 5; // Acceleration in meters per second squared
}